cdk/src/stacks/rootmail.ts

```diff
@@ -1,13 +1,138 @@
-import path from 'path';
-import { NestedStack, NestedStackProps } from 'aws-cdk-lib';
-import { CfnInclude } from 'aws-cdk-lib/cloudformation-include';
+import {
+  Duration,
+  aws_iam as iam,
+  aws_route53 as r53,
+  aws_s3 as s3,
+  aws_ssm as ssm,
+  CfnResource,
+  RemovalPolicy,
+  NestedStack,
+  NestedStackProps,
+  Stack,
+  CfnParameter,
+} from 'aws-cdk-lib';
 import { Construct } from 'constructs';
+import { HostedZoneDkim } from '../constructs/rootmail-hosted-zone-dkim';
+import * as cdk from 'aws-cdk-lib';
+import Fs from 'fs';
+import { CfnRole } from 'aws-cdk-lib/aws-iam';
 
 export class RootmailStack extends NestedStack {
+  public readonly emailBucket: s3.Bucket;
+
   constructor(scope: Construct, id: string, props: NestedStackProps) {
     super(scope, id, props);
-    new CfnInclude(this, 'SuperwerkerTemplate', {
-      templateFile: path.join(__dirname, '..', '..', '..', 'templates', 'rootmail.yaml'),
+
+    const domain = new CfnParameter(this, 'Domain', {
+      type: 'String',
+    });
+
+    const subdomain = new CfnParameter(this, 'Subdomain', {
+      type: 'String',
+      default: 'aws',
+    });
+
+    const totalTimeToWireDNS = new CfnParameter(this, 'TotalTimeToWireDNS', {
+      type: 'Number',
+      default: 480,
+      minValue: 5,
+      maxValue: 480, // 8 hours
+    });
+
+    const propagationParameterName = new CfnParameter(this, 'PropagationParameterName', {
+      type: 'String',
+      default: '/superwerker/propagation_status',
+    });
+
+    const hostedZoneParameterName = new CfnParameter(this, 'HostedZoneParameterName', {
+      type: 'String',
+      default: '/superwerker/domain_name_servers',
+    });
+
+    // Email bucket
+    this.emailBucket = new s3.Bucket(this, 'EmailBucket', {
+      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      encryption: s3.BucketEncryption.S3_MANAGED,
```

Comment:
I assume this does not cause issues on update? since in the original template is not encrypted

Reply:
Yes the existing bucket is just updated & the existing emails are also not affected.